Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -1024,16 +1024,16 @@ void RenderLayer::recursiveUpdateLayerPositions(OptionSet<UpdateLayerPositionsFl
         // LayoutState outside the layout() phase and use it here.
         ASSERT(!renderer().view().frameView().layoutContext().isPaintOffsetCacheEnabled());
 
-        CheckedPtr repaintContainer = renderer().containerForRepaint().renderer;
-        
+        auto* repaintContainer = renderer().containerForRepaint().renderer.get();
+
         auto oldRects = repaintRects();
-        computeRepaintRects(repaintContainer.get());
+        computeRepaintRects(repaintContainer);
         auto newRects = repaintRects();
 
         if (checkForRepaint && shouldRepaintAfterLayout() && newRects) {
             auto needsFullRepaint = m_repaintStatus == RepaintStatus::NeedsFullRepaint ? RequiresFullRepaint::Yes : RequiresFullRepaint::No;
             auto resolvedOldRects = valueOrDefault(oldRects);
-            renderer().repaintAfterLayoutIfNeeded(repaintContainer.get(), needsFullRepaint, resolvedOldRects, *newRects);
+            renderer().repaintAfterLayoutIfNeeded(repaintContainer, needsFullRepaint, resolvedOldRects, *newRects);
```

Comment:
What makes sure that repaintAfterLayoutIfNeeded() doesn't do a use-after-free of `repaintContainer` now?

Reply:
I'd just move it to a lambda to make a scope for repaintContainer and use WeakPtr.